### DISCUSSION TITLE
Filter include using runtime data/related fields

### DISCUSSION THREAD

**Discussion by danshapir** [prisma/prisma#11489] [Q&A] [answered]
How can you filter by part of the query in an include?

Like:
`SELECT * FROM A INNER JOIN B on B.aId = A.id AND B.amount < A.price`

Or:
`SELECT * FROM A INNER JOIN B on a.aId = A.id WHERE B.amount < A.price`

Meaning your query’s filter parameter is part of the query itself.

I couldn’t figure any way to do it and it’s crazy… This is a super basic SQL capability.

**Comment by Luke265** [comment]
I think this is not possible yet. For this you will have to use raw sql. Might be possible to resolve this with Prisma API if Prisma would allow to use partial raw sql, https://github.com/prisma/prisma/issues/5560 , then the solution could look like this:

```js
prisma.a.findMany({
    include: {
       b: true
    },
    where: {
        [raw`b.amount`]: {
            lt: raw`a.price`
        }
    }
});
```

**Comment by pantharshit00** [comment, marked as answer]
This is now possible with https://github.com/prisma/prisma/releases/tag/4.3.0 with field references.

Here is how it will look like:
```ts
prisma.a.findMany({
  include: {
    b: true,
  },
  where: {
    b: {
      amount: {
        lt: prisma.a.fields.amount,
      },
    },
  },
});
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
